openedx/core/djangoapps/theming/tests/test_theme_styling.py

```diff
@@ -0,0 +1,265 @@
+"""
+    Tests for comprehensive themes.
+"""
+import unittest
+
+from django.conf import settings
+from django.test import TestCase
+from django.contrib import staticfiles
+
+from mock import patch
+from paver.easy import call_task
+
+from openedx.core.djangoapps.theming.test_util import with_comprehensive_theme
+
+from pavelib import assets
+
+
+@unittest.skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Test only valid in lms')
+class TestComprehensiveThemeLMS(TestCase):
+    """
+    Test html, sass and static file overrides for comprehensive themes.
+    """
+
+    def setUp(self):
+        """
+        Clear static file finders cache and register cleanup methods.
+        """
+        super(TestComprehensiveThemeLMS, self).setUp()
+        self.addCleanup(self.clean_up)
+
+        # Clear the internal staticfiles caches, to get test isolation.
+        staticfiles.finders.get_finder.cache_clear()
+
+    @classmethod
+    def setUpClass(cls):
+        """
+        Enable Comprehensive theme and compile sass files.
+        """
+        # Apply Comprehensive theme and compile sass assets.
+        with patch("pavelib.assets.Env.env_tokens", {'COMPREHENSIVE_THEME_DIR': settings.TEST_THEME}):
+            # Configure path for themes
+            assets.configure_paths()
+            compile_sass('lms')
+
+        super(TestComprehensiveThemeLMS, cls).setUpClass()
+
+    def clean_up(self):
+        """
+        Disable comprehensive theme and clear changes made for comprehensive themes.
+        """
+        patch("pavelib.assets.Env.env_tokens", {'COMPREHENSIVE_THEME_DIR': ""})
+        clear_theme_sass_dirs()
```

Comment:
I wonder if you should call ``compile_sass`` again to regenerate the test files. If you don't, then someone who runs this test will then have an incorrect css file for any subsequent tests that they run.